packages/wrangler/src/constellation/listCatalog.tsx

```diff
@@ -0,0 +1,73 @@
+import Table from "ink-table";
+import React from "react";
+import { fetchResult } from "../cfetch";
+import { withConfig } from "../config";
+import { logger } from "../logger";
+import { requireAuth } from "../user";
+import { renderToString } from "../utils/render";
+import { constellationBetaWarning } from "./utils";
+import type {
+	CommonYargsArgv,
+	StrictYargsOptionsToInterface,
+} from "../yargs-types";
+import type { CatalogEntry } from "./types";
+
+export function Options(yargs: CommonYargsArgv) {
+	return yargs
+		.option("json", {
+			describe: "return output as clean JSON",
+			type: "boolean",
+			default: false,
+		})
+		.epilogue(constellationBetaWarning);
+}
+
+type HandlerOptions = StrictYargsOptionsToInterface<typeof Options>;
+export const Handler = withConfig<HandlerOptions>(
+	async ({ json, config }): Promise<void> => {
+		const accountId = await requireAuth(config);
+		const entries: Array<CatalogEntry> = await listCatalogEntries(accountId);
+
+		if (json) {
+			logger.log(JSON.stringify(entries, null, 2));
+		} else {
+			logger.log(constellationBetaWarning);
+			logger.log(
+				renderToString(
+					<Table
+						data={entries.map((entry) => ({
+							project_id: entry.project.id,
+							project_name: entry.project.name,
+							project_runtime: entry.project.runtime,
+							models: entry.models.map((model) => model.name).join(","),
+						}))}
+					></Table>
+				)
+			);
+		}
+	}
+);
+
+export const listCatalogEntries = async (
+	accountId: string
+): Promise<Array<CatalogEntry>> => {
+	const pageSize = 10;
+	let page = 1;
+	const results = [];
+	while (results.length % pageSize === 0) {
+		const json: Array<CatalogEntry> = await fetchResult(
```

Comment:
Does this API support larger page sizes than `10`? This recursive calling seems likely to start running into rate limits etc...

Reply:
Changed it to 50.